src/main/java/persistence/sql/dml/SelectQueryBuilder.java

```diff
@@ -22,12 +27,54 @@ public static SelectQueryBuilder getInstance() {
     }
 
     public String generateQuery(Table table) {
-        return String.format(SELECT_DEFINITION, columnsClause(table.getColumns()), table.getTableName());
+        return String.format(SELECT_FORMAT, buildSelectClause(table), buildFromClause(table));
     }
 
-    private String columnsClause(List<Column> columns) {
+    private String buildSelectClause(Table table) {
+        String rootColumns = buildColumnsClause(table.getTableName(), table.getSelectColumns());
+        if (table.getEagerRelationColumns().isEmpty()) {
+            return SELECT_DEFINITION + SPACE.getValue() + rootColumns;
+        }
+        String relationColumns = COMMA.getValue() + buildRelationColumnsClause(table.getEagerRelationColumns());
+        return SELECT_DEFINITION + SPACE.getValue() + rootColumns + relationColumns;
+    }
+
+    private String buildFromClause(Table table) {
+        if (table.getEagerRelationColumns().isEmpty()) {
+            return FROM_DEFINITION + SPACE.getValue() + table.getTableName();
+        }
+        String relationTable = SPACE.getValue() + buildRelationTableClause(table, table.getEagerRelationColumns());
```

Comment:
값을 꺼내서 외부에서 조작하는 일이 많네요 ㅜ
`buildRelationTableClause`함수를 빌더에서 제거하고 객체에게 만들어 달라고 시켜보면 어떨까요? 
지금은 실전이 아니라 연습의 과정이니까 과도하게 객체에게 물어보는것도 좋을것 같아서요!

Reply:
엇 그렇게 되면 table이라는 객체가 쿼리를 만드는 책임을 가져가게 될텐데 
모호하다고 생각이 들게됩니다.

Reply:
그렇네요 책임이 넘어 가버리겠네요. 😂 

```suggestion
        String relationTable = SPACE.getValue() + table.expressionOf(LEFT_JOIN_DEFINITION);
```

코멘트 남길시기에는 이렇게 생각했었는데.
생각보다 `LEFT_JOIN_DEFINITION`의 규칙이 복잡 단단하군요... ㅜㅜ
Table이 `LEFT_JOIN_DEFINITION` 종속적인 기능이 되어버릴것 같아 처음 생각했던 방식은 좋은 방법이 아니라고 이제는 생각되네요. 😂

Reply:
`LEFT_JOIN_DEFINITION` 규칙에 많은 책임이 있는것 같아 다소 복잡해 보이긴해요. 🐱 

---

`LEFT_JOIN_DEFINITION = "LEFT JOIN %s ON %s = %s";`
or
`LEFT_JOIN_DEFINITION = "LEFT JOIN %s ON %s";`

음... 데이터를 합치는`(tableName.columnName)` 책임을 객체에게 위임시키는 방법에 대해서 어떻게 생각하시나요~?

Reply:
```suggestion
        String relationTable = SPACE.getValue() + buildRelationTableClause(table);
```
table 에서 미리 꺼내어 인자갯수를 안늘려도 되지않을까요? 🤔